File: dotnet/Lexy/Lexy.RunTime/RunTime/BuiltInNumberFunctions.cs
using System;

namespace Lexy.RunTime.RunTime
{
    public static class BuiltInNumberFunctions
    {
        public static decimal Int(decimal value) => Math.Floor(value);
        public static decimal Abs(decimal value) => Math.Abs(value);
        public static decimal Power(decimal number, decimal power) => (decimal)Math.Pow((double) number, (double) power);
        public static decimal Round(decimal number, decimal digits) => Math.Round(number, (int) digits);
    }
}

File: typescipt/src/runTime/builtInNumberFunctions.ts


export class BuiltInNumberFunctions {
   public static number(value: decimal): decimal {
     return Math.Floor(value);
   }

   public static abs(value: decimal): decimal {
     return Math.Abs(value);
   }

   public static power(number: decimal, power: decimal): decimal {
     return (decimal)Math.Pow((double)number, (double)power);
   }

   public static round(number: decimal, digits: decimal): decimal {
     return Math.Round(number, (number)digits);
   }
}
